Plugin/AddToCartRedirectForInsights.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Plugin;
+
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use Magento\Checkout\Model\Cart;
+use Magento\Checkout\Model\Cart\RequestInfoFilterInterface;
+use Magento\Checkout\Model\Session;
+use Magento\Framework\DataObject;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+
+class AddToCartRedirectForInsights
+{
+    /**
+     * @var RequestInfoFilterInterface
+     */
+    private $requestInfoFilter;
+
+    public function __construct(
+        protected StoreManagerInterface $storeManager,
+        protected ProductRepositoryInterface $productRepository,
+        protected Session $checkoutSession,
+        protected StockRegistryInterface $stockRegistry,
+        protected ManagerInterface $eventManager,
+        protected ConfigHelper $configHelper,
+    ) {}
+
+    /**
+     * @param Cart $cartModel
+     * @param int|Product $productInfo
+     * @param array|int|DataObject|null $requestInfo
+     *
+     * @return void
```

Comment:
Ideally for "before" plugins [we should return `null`](https://developer.adobe.com/commerce/php/development/components/plugins/#before-methods) if we are not modifying the arguments. 

[Since `null` is returned anyway](https://www.php.net/manual/en/functions.returning-values.php) when return is omitted this code should work without issue. However, for best practice we should properly docblock this:

```suggestion
     * @return null
```